test/integration/metrics/metrics_test.go

```diff
@@ -241,3 +241,255 @@ func TestApiserverMetricsLabels(t *testing.T) {
+						Image: "image",
+					},
+				},
+			},
+		}
+	}
+
+	_, s, closeFn := framework.RunAMaster(framework.NewMasterConfig())
+	defer closeFn()
+
+	client, err := clientset.NewForConfig(&restclient.Config{Host: s.URL, QPS: -1})
+	if err != nil {
+		t.Fatalf("apiserver_request_total metric not exposed")
+	}
+
+	c := client.CoreV1().Pods(metav1.NamespaceDefault)
+
+	for _, tc := range []struct {
+		name     string
+		executor func()
+
+		want string
+	}{
+		{
+			name: "create pod",
+			executor: func() {
+				callOrDie(c.Create(context.TODO(), makePod("foo"), metav1.CreateOptions{}))
+			},
+			want: `apiserver_request_total{code="201", component="apiserver", contentType="application/json", dry_run="", group="", resource="pods", scope="resource", subresource="", verb="POST", version="v1"} => 1 @[0]`,
+		},
+		{
+			name: "update pod",
+			executor: func() {
+				callOrDie(c.Update(context.TODO(), makePod("bar"), metav1.UpdateOptions{}))
+			},
+			want: `apiserver_request_total{code="200", component="apiserver", contentType="application/json", dry_run="", group="", resource="pods", scope="resource", subresource="", verb="PUT", version="v1"} => 1 @[0]`,
+		},
+		{
+			name: "update pod status",
+			executor: func() {
+				callOrDie(c.UpdateStatus(context.TODO(), makePod("bar"), metav1.UpdateOptions{}))
+			},
+			want: `apiserver_request_total{code="200", component="apiserver", contentType="application/json", dry_run="", group="", resource="pods", scope="resource", subresource="status", verb="PUT", version="v1"} => 1 @[0]`,
+		},
+		{
+			name: "get pod",
+			executor: func() {
+				callOrDie(c.Get(context.TODO(), "foo", metav1.GetOptions{}))
+			},
+			want: `apiserver_request_total{code="200", component="apiserver", contentType="application/json", dry_run="", group="", resource="pods", scope="resource", subresource="", verb="GET", version="v1"} => 1 @[0]`,
+		},
+		{
+			name: "list pod",
+			executor: func() {
+				callOrDie(c.List(context.TODO(), metav1.ListOptions{}))
+			},
+			want: `apiserver_request_total{code="200", component="apiserver", contentType="application/json", dry_run="", group="", resource="pods", scope="namespace", subresource="", verb="LIST", version="v1"} => 1 @[0]`,
+		},
+		{
+			name: "delete pod",
+			executor: func() {
+				callOrDie(nil, c.Delete(context.TODO(), "foo", metav1.DeleteOptions{}))
+			},
+			want: `apiserver_request_total{code="200", component="apiserver", contentType="application/json", dry_run="", group="", resource="pods", scope="resource", subresource="", verb="DELETE", version="v1"} => 1 @[0]`,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+
+			baseSamples, err := getSamples(s)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			tc.executor()
+
+			updatedSamples, err := getSamples(s)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			newSamples := diffMetrics(updatedSamples, baseSamples)
+			found := false
+
+			for _, s := range newSamples {
+				if s.String() == tc.want {
+					found = true;
```

Comment:
To pass pull-kubernetes-verify job, the following change seems necessary:
```
diff -u ./test/integration/metrics/metrics_test.go.orig ./test/integration/metrics/metrics_test.go
--- ./test/integration/metrics/metrics_test.go.orig	2020-11-03 17:18:04.276518016 +0000
+++ ./test/integration/metrics/metrics_test.go	2020-11-03 17:18:04.276518016 +0000
@@ -344,7 +344,7 @@
 
 			for _, s := range newSamples {
 				if s.String() == tc.want {
-					found = true;
+					found = true
 					break
 				}
 			}
@@ -450,7 +450,7 @@
 
 			for _, s := range newSamples {
 				if s.String() == tc.want {
-					found = true;
+					found = true
 					break
 				}
 			}

Run ./hack/update-gofmt.sh
```

Reply:
Thanks! added the code.